telemetry/client.go

```diff
@@ -157,11 +152,14 @@ func (c *Client) processAndPushTelemetry(data interface{}) {
 		return
 	}
 
-	c.nextId++
 	c.telemetryCh <- telemetryRequest
+	c.nextIdLock.Lock()
+	c.nextId++
+	c.nextIdLock.Unlock()
 }
 
 func (c *Client) pushTelemetryRequest(request []TelemetryRequest) {
+	c.logger.Debug("Publishing to telemetry", zap.Int("msgNum", len(request)))
```

Comment:
When I added something like this in my last PR I was told it would make logs noisy as this is an expected result (at least when telemetry is turned on). After using status desktop and inspecting logs locally, I agree, as there is already quite a lot being logged.